lib/model/migrations/20230406-01-add-entity-def-fields.js

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 ODK Central Developers
+// See the NOTICE file at the top-level directory of this distribution and at
+// https://github.com/getodk/central-backend/blob/master/NOTICE.
+// This file is part of ODK Central. It is subject to the license terms in
+// the LICENSE file found in the top-level directory of this distribution and at
+// https://www.apache.org/licenses/LICENSE-2.0. No part of ODK Central,
+// including this file, may be copied, modified, propagated, or distributed
+// except according to the terms contained in the LICENSE file.
+
+const up = async (db) => {
+  // Add columns
+  await db.raw(`
+ALTER TABLE entity_defs
+ADD COLUMN "creatorId" integer,
+ADD COLUMN "userAgent" varchar(255)
+`);
+
+  // Assign submission submitter id to entity def creator id
+  await db.raw(`
+UPDATE entity_defs
+set "creatorId" = submission_defs."submitterId",
+"userAgent" = submission_defs."userAgent"
+FROM submission_defs
+WHERE entity_defs."submissionDefId" = submission_defs.id;
+  `);
+};
```

Comment:
I think we should alter `entity_defs` table again here to make new columns `NOT NULL`

Reply:
I think `creatorId` should be `NOT NULL`, but I don't think `userAgent` can be. I think the user agent is optional in the request (it's `NOT NULL` on the `submissions` table). Also, the submission might have been deleted, in which case the user agent will be gone.

Reply:
Take a look at my last 2 commits where I change up the migrations a bit. Now the new `creatorId` column is NOT NULL (and it come from the entity rather than submission def) and `userAgent` can still be NULL.

Reply:
did you forget to remove `ALTER COLUMN "userAgent" SET NOT NULL`?